lib/dealer.rb

```diff
@@ -1,55 +1,68 @@
 require_relative 'blackjack_score'
 require_relative 'hand'
+require_relative 'commandlineio'
 
 class Dealer
 
-	def initialize(output = $stdout, input = $stdin)
-		@output = output
-		@input = input
-	end
-
-	def ask_hit_or_stop
-		@output.puts "Do you want to hit or stop?"
-	end
-
-	def get_hit_or_stop
-		user_input = @input.gets.chomp.downcase
-		until user_input == "hit" || user_input == "stop"
-			puts "Please enter hit or stop: "
-			user_input = @input.gets.chomp.downcase
-		end
-		user_input
+	def initialize(io = CommandlineIO.new)
+		@io = io
 	end
 
 	def give_new_card
 		card_values = ("1".."10").to_a + ["J", "Q", "K", "A"]
 		card_type = ["C", "D", "H", "S"]
 		new_card = [card_values.sample + card_type.sample]
```

Comment:
This could, theoretically, give the same card twice.  There should only be 52 cards in a deck.  Please could you change this so that this is no longer possible?